widgets/static/audioProcessor.js

```diff
@@ -0,0 +1,38 @@
+class AudioDataExtractor extends AudioWorkletProcessor {
+    _updateIntervalInMS;
+    _sampleInFrames;
+    _index;
+    _buffer;
+
+    constructor() {
+        super();
+        this._updateIntervalInMS = 50;
```

Comment:
This is a bit too small IMO. It means that audio chunks are only 0.05 seconds long no? Think we should increase this to 0.2 seconds. In my experiments, the quality was significantly better when increasing the chunk length and a delay of 0.2 is still very good!